frontend/src/js/components/deployments/deployment-wizard/ReleaseArtifactFilter.tsx

```diff
@@ -0,0 +1,162 @@
+// Copyright 2025 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+import { useEffect, useRef, useState } from 'react';
+import { FormProvider, useForm, useWatch } from 'react-hook-form';
+import { useSelector } from 'react-redux';
+
+import { Close as CloseIcon, ExpandLess as ExpandLessIcon, ExpandMore as ExpandMoreIcon, FilterList as FilterListIcon } from '@mui/icons-material';
+import { Button, DialogContent, Divider, TextField, Typography } from '@mui/material';
+import { makeStyles } from 'tss-react/mui';
+
+import ChipSelect from '@northern.tech/common-ui/ChipSelect';
+import { ControlledSearch } from '@northern.tech/common-ui/Search';
+import { BaseDialog } from '@northern.tech/common-ui/dialogs/BaseDialog';
+import { ControlledAutoComplete } from '@northern.tech/common-ui/forms/Autocomplete';
+import { Release } from '@northern.tech/store/releasesSlice';
+import { getReleaseListState, getReleaseTags, getReleasesById, getUpdateTypes } from '@northern.tech/store/selectors';
+import { useAppDispatch } from '@northern.tech/store/store';
+import { getReleases } from '@northern.tech/store/thunks';
+import { useDebounce } from '@northern.tech/utils/debouncehook';
+
+import { ReleaseItem } from './ReleaseItem';
+
+const useStyles = makeStyles()(theme => ({
+  advancedFilterContainer: {
+    width: 'calc(50% - 8px)'
+  },
```

Comment:
that to me looks a lot like what our `.two-columns` class does/ what a grid implementation would be better suited for...